pkg/kubelet/kubelet.go

```diff
@@ -901,6 +901,23 @@ func (kl *Kubelet) initialNodeStatus() (*api.Node, error) {
 		if err != nil {
 			return nil, err
 		}
+
+		// If the cloud has zone information, label the node with the zone information
+		zones, ok := kl.cloud.Zones()
```

Comment:
Can we have a phase where the cluster is still not `logically` up? I'd imagine moving logging and monitoring agents to be helper pods (daemonSet) in the future and the cluster isn't logically usable until those pods come up.

@quinton-hoole: My concern is that this PR will open up a doorway for exposing more and more labels from the kubelet directly. There are many other use-cases already where people want the node to expose custom labels.
From core kubernetes perspective, I see these labels are being extraneous. The cluster can function without them. 

Existing issues related to node labels: #13524 #13520